src/main/java/teammates/logic/core/MailjetService.java

```diff
@@ -0,0 +1,62 @@
+package teammates.logic.core;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+
+import com.mailjet.client.MailjetClient;
+import com.mailjet.client.MailjetRequest;
+import com.mailjet.client.MailjetResponse;
+import com.mailjet.client.errors.MailjetException;
+import com.mailjet.client.resource.Email;
+
+import teammates.common.util.Config;
+import teammates.common.util.EmailWrapper;
+
+/**
+ * Email sender service provided by Mailjet.
+ * Reference: https://cloud.google.com/appengine/docs/java/mail/mailjet
+ * 
+ * @see MailjetClient
+ * @see MailjetRequest
+ * @see MailjetResponse
+ */
+public class MailjetService implements EmailSenderService {
+    
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public MailjetRequest parseToEmail(EmailWrapper wrapper) {
+        MailjetRequest request = new MailjetRequest(Email.resource);
+        request.property(Email.FROMEMAIL, wrapper.getSenderEmail());
+        if (wrapper.getSenderName() != null && !wrapper.getSenderName().isEmpty()) {
+            request.property(Email.FROMNAME, wrapper.getSenderName());
+        }
+        
+        request.property(Email.RECIPIENTS, new JSONArray().put(new JSONObject().put("Email", wrapper.getRecipient())));
+        if (wrapper.getBcc() != null && !wrapper.getBcc().isEmpty()) {
+            request.property(Email.BCC, new JSONArray().put(new JSONObject().put("Email", wrapper.getBcc())));
```

Comment:
Thanks for this. I didn't test the BCC features before, but just now I tested on all services and confirmed that all is well.
Mailjet is a strange case though, they somehow made the "Recipients" field mandatory even though they explicitly said that BCC can't be used in conjunction with that (in other words, replacing "Recipients" with "To" does not work because there will be a missing field). As a workaround, the BCC is appended to the "Recipients" field. This does not disclose the identity in the BCC field because every email will be stamped with one and only one recipient.